src/runtime_src/core/tools/xbtracer/src/app/launcher.cpp

```diff
@@ -0,0 +1,389 @@
+  app.m_si.cb = sizeof(STARTUPINFOA);
+  ZeroMemory(&app.m_pi, sizeof(PROCESS_INFORMATION));
+
+  if (CreateProcessA(
+      NULL,                         // No module name (use command line)
+      (LPSTR)app.m_cmdline.c_str(), // Command line
+      NULL,                         // Process handle not inheritable
+      NULL,                         // Thread handle not inheritable
+      FALSE,                        // Set handle inheritance to FALSE
+      CREATE_SUSPENDED,             // Process created in a suspended state
+      NULL,                         // Use parent's environment block
+      NULL,                         // Use parent's starting directory
+      &app.m_si,                    // Pointer to STARTUPINFO structure
+      &app.m_pi) == FALSE)          // Pointer to PROCESS_INFORMATION structure
+  {
+    log_f("Child process creation failed");
+    return -1;
+  }
+
+  log_d("Child process created. ");
+
+  return 0;
+
+}
+
+int resume_child_proc_and_wait_for_completion(launcher& app)
+{
+  DWORD retval = 0;
+  if (ResumeThread(app.m_pi.hThread) < 0) {
+    log_e("Failed to resume thread");
+    return -1;
+  }
+
+  // Wait for child process to finish
+  if (WaitForSingleObject(app.m_pi.hProcess, INFINITE) != WAIT_OBJECT_0) {
+    log_e("Waiting for child process failed");
+    return -2;
+  }
+  log_d("Child process resumed, Waiting for child process to finish");
+
+  // Get return code and forward it
+  if (GetExitCodeProcess(app.m_pi.hProcess, &retval) == FALSE) {
+    log_e("Failed to read child process exit code");
+    return -3;
+  }
+
+  return retval;
+}
+
+int win_launcher(int& argc, char* argv[])
+{
+  DWORD retval = 0;
+  launcher& app = launcher::get_instance();
+
+  std::filesystem::path path(argv[0]);
+  if (!path.filename().empty())
+    app.m_name = path.filename().string();
+
+  /**
+    Parse arguments
+  */
+  parse_cmdline(app, argc, argv);
+
+  /**
+    Create child process as suspended
+  */
+  create_child_proc_as_suspended(app);
+
+  /*
+    Check compatibility between the child and parrent.
+  */
+  if (check_compatibility(GetCurrentProcess(), app.m_pi.hProcess) == false)
+    log_f("Compatability check failed. Exiting ...");
+
+  /**
+    Resume child process and wait for finish
+  */
+  log_d("Resuming child process");
+  retval = resume_child_proc_and_wait_for_completion(app);
+
+  log_d("Child process completed with exit code ", retval);
+
+  return 0;
+}
+#else // #ifdef _WIN32
+int posix_launcher(int& argc, char* argv[])
+{
+  launcher& app = launcher::get_instance();
+
+  std::filesystem::path path(argv[0]);
+  if (!path.filename().empty())
+    app.m_name = path.filename().string();
+
+  /**
+    Parse arguments
+  */
+  parse_cmdline(app, argc, argv);
+
+  char *command_args[max_cmd_args];
+
+  convert_to_c_array(app.m_child_cmd_args, command_args, max_cmd_args);
+
+  execve(app.m_child_cmd_args.front().data(), command_args, (char* const*)environ);
```

Comment:
Nit, here I would have overwritten execve and done the conversion to c_array in the overwrite before calling system execve.   Reason for this is that the c-array is a limitation of execve not of posix_launcher.

Reply:
Not sure, if I completely understood. I have wrapped all of these in separate function named `launch_process` and invoked it from posix_launcher. Please let me know if I missed anything.